src/modules/RokulRunnings.ts

```diff
@@ -175,13 +230,45 @@ export class RokulRunnings {
     delayInMillis = this.pressDelayInMillis,
     retries = this.retries,
     params,
+    keyType = 'press',
   }: {
     sequence: Buttons[] | string[];
     delayInMillis?: number;
     retries?: number;
     params?: Params;
+    keyType?: 'up' | 'down' | 'press';
+  }) {
+    const newSequence: { up?: string | Buttons; down?: string | Buttons; press?: string | Buttons }[] = [];
+    sequence.forEach(button => {
+      if (!newSequence) {
+        newSequence[0] = { [keyType]: button };
+      } else {
+        newSequence.push({ [keyType]: button });
+      }
+    });
+    const response = await this.driver.sendSequence({ sequence: newSequence, delayInMillis, params, retries });
+    const formattedResponse: { [key: string]: number }[] = [];
+    response.forEach(responseObject => {
+      const key = Object.keys(responseObject)[0];
+      const newKey = key.replace(`${keyType}:`, '');
+      formattedResponse.push({ [newKey]: responseObject[key] });
+    });
+    return formattedResponse;
+  }
+
+  /** Allows for a sequence to be sent that uses keyUp, keyDown, and keyPress */
+  async sendMixedButtonSequence({
+    customSequence,
+    delayInMillis = this.pressDelayInMillis,
+    retries = this.retries,
+    params,
+  }: {
+    customSequence: [{ up?: string; down?: string; press?: string }];
```

Comment:
I think this definition allows all 3 at once, and also none.

Reply:
How would I make the definition require one, and only allow one?

Reply:
I think `{ up: string; } | { down: string; } | { press: string }[]` does what you want.

Reply:
Resolved in eee2470